.github/actions/wait-for-check-completion/src/index.js

```diff
@@ -0,0 +1,194 @@
+
+    const checkResults = {};
+    const pendingChecks = new Set(checkNames);
+
+    while ((Date.now() - startTime < timeoutMs) && (pendingChecks.size > 0)) {
+        try {
+            // Get all check runs for the specific commit
+            const { data: checkRuns } = await octokit.rest.checks.listForRef({
+                owner,
+                repo,
+                ref,
+                per_page: 100
+            });
+
+            core.info(`Found ${checkRuns.check_runs.length} total check runs`);
+
+            // Process each pending check
+            for (const checkName of Array.from(pendingChecks)) {
+                const matchingRuns = checkRuns.check_runs.filter(run => run.name === checkName);
+
+                if (matchingRuns.length === 0) {
+                    core.info(`No check runs found for "${checkName}" yet, waiting...`);
+                    continue;
+                }
+
+                // Find the most recent check run for this check
+                const latestCheckRun = matchingRuns.sort((a, b) =>
+                    new Date(b.started_at) - new Date(a.started_at)
+                )[0];
+
+                core.info(`Check "${checkName}" status: ${latestCheckRun.status}, conclusion: ${latestCheckRun.conclusion || 'N/A'}`);
+
+                if (latestCheckRun.status === 'completed') {
+                    core.info(`Check "${checkName}" completed with conclusion: ${latestCheckRun.conclusion}`);
+                    checkResults[checkName] = {
+                        status: latestCheckRun.status,
+                        conclusion: latestCheckRun.conclusion,
+                        checkRun: latestCheckRun
+                    };
+                    pendingChecks.delete(checkName);
+                } else if (latestCheckRun.status === 'in_progress') {
+                    core.info(`Check "${checkName}" is still in progress...`);
+                } else if (latestCheckRun.status === 'queued') {
+                    core.info(`Check "${checkName}" is queued...`);
+                }
+            }
+
+            if (pendingChecks.size === 0) {
+                core.info('All checks completed, parsing conclusions...');
+                break;
+            }
+
+        } catch (error) {
+            core.warning(`Error fetching check runs: ${error.message}`);
+
+            // If it's a 404, the checks might not exist yet
+            if (error.status === 404) {
+                core.info('Checks not found yet, continuing to wait...');
+            } else {
+                // For other errors
+                core.error(`API error: ${error.message}`);
+            }
+        }
+
+        if (pendingChecks.size > 0) {
+            core.info(`Still waiting for [${Array.from(pendingChecks).join(', ')}]. Waiting ${waitInterval} seconds before next check...`);
+            await new Promise(resolve => setTimeout(resolve, waitIntervalMs));
+        }
```

Comment:
Can we move this if-statement inside try block and combine (add else branch) with if statement where `core.info('All checks completed, parsing conclusions...');` is logging?